modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.tsx

```diff
@@ -133,12 +139,48 @@ const FrontendDataSetContent = ({
 	showSelectAll = false,
 	sidePanelId,
 	sorts: sortsProp = [],
+	stateInURLSettings = EStateInURLSettings.OFF,
 	style = 'default',
 	uniformActionsDisplay,
 	views,
 }: IFrontendDataSetProps) => {
 	const fdsRef = useRef(null);
 	const dataSetWrapperRef: RefObject<HTMLDivElement> = useRef(null);
+	const getStateFromURL = useGetStateFromURL({
+		id,
+		stateInitializers: {
+			[EStateInURLKeys.DELTA]: (delta: number) => {
+				if (isNaN(delta) || delta < 1) {
+					return null;
+				}
+
+				return delta;
+			},
+			[EStateInURLKeys.VIEW_NAME]: (viewName: string) => {
+				const view = views.find(({name}) => name === viewName);
+
+				if (view) {
+					return viewName;
+				}
+
+				return null;
+			},
+		},
+	});
+	const stateInURLSetters = useSetStateInURL({
+		id,
+		setters: [
+			{
+				key: EStateInURLKeys.DELTA,
+				type: VIEWS_ACTION_TYPES.UPDATE_PAGINATION_DELTA,
+			},
+			{
+				key: EStateInURLKeys.VIEW_NAME,
+				type: VIEWS_ACTION_TYPES.UPDATE_ACTIVE_VIEW,
+			},
+		],
```

Comment:
This is ok, but I find it odd to have type of action as a property. If the idea is reuse update actions in batch action, would it simplify code to have just `key` here andbranch internally in reducer? Something like 
```js
// in loop
if (key === EStateInURLKeys.DELTA) {
    VIEWS_ACTION_TYPES.UPDATE_PAGINATION_DELTA(value)
} else if (key === EStateInURLKeys.VIEW_NAME) {
    VIEWS_ACTION_TYPES.UPDATE_ACTIVE_VIEW(value)
}
```

Reply:
These setters are not using a batch action, they just work isolatedly. Each setter function returned by this hook is in charge of dispatching a single type of action and writing the state in the URL at the same time. So we have one for delta, other for views, and more will come if we adopt this pattern.

Perhaps we can simplify this by providing the array of setters as an object:
``` js
setters: {
	[EStateInURLKeys.DELTA]: VIEWS_ACTION_TYPES.UPDATE_PAGINATION_DELTA,
	[EStateInURLKeys.VIEW_NAME]: VIEWS_ACTION_TYPES.UPDATE_ACTIVE_VIEW,
},
```

How does that sound?